src/Compilers/CSharp/Test/Syntax/LexicalAndXml/XmlDocCommentTests.cs

```diff
@@ -3269,6 +3269,31 @@ public void WriteDocumentationCommentXml_01()
                 );
         }
 
+        [WorkItem(47278, "https://github.com/dotnet/roslyn/issues/47278")]
+        [Fact]
+        public void NormalizeWhitespace()
+        {
+            var text = @"
+namespace Foo
+{
+    /// <summary>
+    /// This is Foo.Bar
+    /// </summary>
+    class Bar { }
+}";
+
+            var tree = Parse(text);
+            tree = tree.WithRootAndOptions(tree.GetRoot().NormalizeWhitespace(), tree.Options);
```

Comment:
Consider verifying the `ToFullString` of this tree. I'm not clear on what bug this actually fixing.

Reply:
It's not important that `NormalizeWhitespace` changes the text here (and I have added a check verifying that it doesn't). What's important is how it changes the structure of the tree. Specifically, before `NormalizeWhitespace`, `tree.GetRoot().DescendantTrivia().Single(n => n.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia)).ToFullString()` returns:

```
"/// <summary> \n    /// This is Foo.Bar \n    /// </summary> \n"
```

But after the call, it's:

```
"/// <summary> \n    /// This is Foo.Bar \n    /// </summary> \n    "
```

I haven't looked into why `NormalizeWhitespace` does this and changing it would be another way to fix this issue, but I think making `DocumentationCommentCompiler` more robust (especially since it's such a simple change) is the better option here.

Reply:
This was important context. Please verify the `ToFullString` of the text here to make that context apparent in the test itself :)

Reply:
@333fred I have added some asserts. Is that what you had in mind?